args.go

```diff
@@ -42,6 +43,9 @@ func newArgBuilder(opts ...Arg) (*argBuilder, error) {
 
 	var buildErr error
 	for _, opt := range opts {
+		if opt == nil {
+			return nil, fmt.Errorf("arg cannot be nil")
```

Comment:
Super nit: we're not doing any error formatting here so maybe use `errors.New()` here.